flow/Trace.cpp

```diff
@@ -220,6 +220,34 @@ struct TraceLog {
 		}
 	};
 
+	struct IssuesList : ITraceLogIssuesReporter, ThreadSafeReferenceCounted<IssuesList> {
+		IssuesList() : moved(false){};
+		void addIssue(std::string issue) override {
+			MutexHolder h(mutex);
+			if (moved) {
+				issues = std::set<std::string>();
+				moved = false;
+			}
+			issues.insert(issue);
+		}
+
+		std::set<std::string> getAndFlushIssues() override {
```

Comment:
One thing I'm now wondering is whether it would be helpful to not flush issues immediately, but instead give them an expiration or require some sort of positive evidence that the problem no longer exists (e.g. that we were able to open the file or successfully write). Just throwing the thought out there.

Reply:
This is implemented now. Issues will now expire after 5 seconds(controlled via knob)